projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_base.py

```diff
@@ -129,6 +130,16 @@ def send_object_for_ingestion(
         """
         See parent doc
         """
+        if self.__object_is_data_frame(payload):
+            return self.send_tabular_data_for_ingestion(
+                rows=payload.itertuples(index=False),
```

Comment:
We discussed that pandas is not thread safe. 

As ingester_base is executed in a background thread. Is this safe? And why is it?

Reply:
I don't understand this comment. He is passing in an iterator when the function says it takes in an iterator. 
Why is there a concern it is not safe?

Can you elaborate on this?

Reply:
@murphp15 since we aren't copying the df, if we access it while its being ingested, it could cause inconsistencies; but I don't understand which user would access and change an object while its being ingested, is that even possible @antoniivanov

Reply:
I'm happy to approve. 
That seems totally reasonable.

Reply:
> but I don't understand which user would access and change an object while its being ingested, is that even possible @antoniivanov

While it's certainly possible ...

```
job_input.send_object_for_ingestion(df, ..) 
df.append("new_data") 

```

... I cannot see many users doing it. Most users would collect the data into data frame and finally send it for ingestion. 

Maybe if they loop
```
for page in pages: 
  # it would be more natural to do df = page. But maybe someone wants to optimize it. 
   df.clear();
  df.append(page) 
  send ...
 ``` 
 
 Let's document this in the API description (job_input API) which states that users should not try to modify the data frame once it's passed to job_input.send ...